From 5c912fcfaff2c7c34232562a118fa5722c6df995 Mon Sep 17 00:00:00 2001
From: Martin Cermak <mcermak@redhat.com>
Date: Fri, 8 Aug 2025 23:46:53 +0200
Subject: [PATCH 3/3] Add several missing syscall hooks to ppc64-linux

Specifically:

sys_getrlimit
sys_mincore
sys_tkill
sys_unshare
sys_splice
sys_tee
sys_vmsplice
sys_fanotify_init
sys_fanotify_mark
sys_kcmp
sys_bpf

https://bugs.kde.org/show_bug.cgi?id=508030
(cherry picked from commit 4363549490ecaeaa6c3048beeb8c5edcec33ec06)
---
 NEWS                                      |  1 +
 coregrind/m_syswrap/syswrap-ppc64-linux.c | 14 +++++++++++---
 2 files changed, 12 insertions(+), 3 deletions(-)

diff --git a/NEWS b/NEWS
index 4e6e6c6d2387..3eea6bb1b709 100644
--- a/NEWS
+++ b/NEWS
@@ -6,6 +6,7 @@ Branch 3.25.x
 The following bugs have been fixed or resolved on this branch.
 
 503241  s390x: Support z17 changes to the NNPA instruction
+508030  Add several missing syscall hooks to ppc64-linux
 
 To see details of a given bug, visit
   https://bugs.kde.org/show_bug.cgi?id=XXXXXX
diff --git a/coregrind/m_syswrap/syswrap-ppc64-linux.c b/coregrind/m_syswrap/syswrap-ppc64-linux.c
index 007fa6336c0e..03a5e6a822c1 100644
--- a/coregrind/m_syswrap/syswrap-ppc64-linux.c
+++ b/coregrind/m_syswrap/syswrap-ppc64-linux.c
@@ -691,7 +691,7 @@ static SyscallTableEntry syscall_table[] = {
 // _____(__NR_sethostname,       sys_sethostname),        //  74
 
    GENX_(__NR_setrlimit,         sys_setrlimit),          //  75
-// _____(__NR_getrlimit,         sys_getrlimit),          //  76
+   GENXY(__NR_getrlimit,         sys_getrlimit),          //  76
    GENXY(__NR_getrusage,         sys_getrusage),          //  77
    GENXY(__NR_gettimeofday,      sys_gettimeofday),       //  78
 // _____(__NR_settimeofday,      sys_settimeofday),       //  79
@@ -847,9 +847,9 @@ static SyscallTableEntry syscall_table[] = {
    LINXY(__NR_fcntl64,           sys_fcntl64),            // 204 !!!!?? 32bit only */
 
    GENX_(__NR_madvise,           sys_madvise),            // 205
-// _____(__NR_mincore,           sys_mincore),            // 206
+   GENXY(__NR_mincore,           sys_mincore),            // 206
    LINX_(__NR_gettid,            sys_gettid),             // 207
-// _____(__NR_tkill,             sys_tkill),              // 208
+   LINXY(__NR_tkill,             sys_tkill),              // 208
    LINX_(__NR_setxattr,          sys_setxattr),           // 209
 
    LINX_(__NR_lsetxattr,         sys_lsetxattr),          // 210
@@ -934,6 +934,10 @@ static SyscallTableEntry syscall_table[] = {
 
    LINXY(__NR_pselect6,          sys_pselect6),           // 280
    LINXY(__NR_ppoll,             sys_ppoll),              // 281
+   LINX_(__NR_unshare,           sys_unshare),            // 282
+   LINX_(__NR_splice,            sys_splice),             // 283
+   LINX_(__NR_tee,               sys_tee),                // 284
+   LINXY(__NR_vmsplice,          sys_vmsplice),           // 285
 
    LINXY(__NR_openat,            sys_openat),             // 286
    LINX_(__NR_mkdirat,           sys_mkdirat),            // 287
@@ -972,6 +976,8 @@ static SyscallTableEntry syscall_table[] = {
    LINXY(__NR_preadv,            sys_preadv),           // 320
    LINX_(__NR_pwritev,           sys_pwritev),          // 321
    LINXY(__NR_rt_tgsigqueueinfo, sys_rt_tgsigqueueinfo),// 322
+   LINXY(__NR_fanotify_init,     sys_fanotify_init),    // 323
+   LINX_(__NR_fanotify_mark,     sys_fanotify_mark),    // 324
 
    LINXY(__NR_prlimit64,         sys_prlimit64),        // 325
    LINXY(__NR_socket,            sys_socket),           // 326
@@ -1001,6 +1007,7 @@ static SyscallTableEntry syscall_table[] = {
 
    LINXY(__NR_process_vm_readv,  sys_process_vm_readv), // 351
    LINX_(__NR_process_vm_writev, sys_process_vm_writev),// 352
+   LINX_(__NR_kcmp,              sys_kcmp),             // 354
 
    LINX_(__NR_sched_setattr,     sys_sched_setattr),    // 355
    LINXY(__NR_sched_getattr,     sys_sched_getattr),    // 356
@@ -1008,6 +1015,7 @@ static SyscallTableEntry syscall_table[] = {
 
    LINXY(__NR_getrandom,         sys_getrandom),        // 359
    LINXY(__NR_memfd_create,      sys_memfd_create),     // 360
+   LINXY(__NR_bpf,               sys_bpf),              // 361
 
    LINX_(__NR_execveat,          sys_execveat),         // 362
 
-- 
2.50.1

